src/mam4xx/gasaerexch.hpp

```diff
@@ -33,7 +33,8 @@ class GasAerExch {
   static constexpr int num_aer = AeroConfig::num_aerosol_ids();
   static constexpr int nait = static_cast<int>(ModeIndex::Aitken);
   static constexpr int npca = static_cast<int>(ModeIndex::PrimaryCarbon);
-  static constexpr int igas_h2so4 = static_cast<int>(GasId::H2SO4);
+  static constexpr int igas_h2so4 = 1;
```

Comment:
Should we modify `GasId`? The current implementation has `GasId::H2SO4` as 2. I would change this enum class to reflect this change.

```
// Identifiers for gas species in MAM4
enum class GasId {
  O3 = 0,    // ozone
  H2O2 = 1,  // hydrogen peroxide
  H2SO4 = 2, // sulfuric acid
  SO2 = 3,   // sulfur dioxide
  DMS = 4,   // dimethyl sulfide
  SOAG = 5,  // secondary organic aerosol precursor
  None = 6,  // invalid gas id
};
```
These hard-coded indices can become a problem for us later when we modify the gas chemistry.

Reply:
Let me verify that. Amicphys has its own indexing and it uses only two gases. We may want to use this special indexing only for amicphys. 

I agree that this hard coding is problematic. We will address it in our code enhancement tasks next year.

Reply:
Gas aerosol exchange uses just two gases: H2SO4 and SOAG. I have now removed all the unused indices in this file. We do not need to change anything else as these index values are specific to amicphys only.